File: bindings/python/insight/signal/bsplines.py
"""B-spline basis functions for signal processing.

Provides Gaussian, cubic, and quadratic B-spline evaluation functions.
"""

from insight._insight import signal as _signal

__all__ = [
    "gauss_spline",
    "cubic",
    "quadratic",
]


def gauss_spline(x, n):
    """Return a Gaussian B-spline basis function.

    Evaluates the *n*-th order Gaussian B-spline at the points in *x*.

    Args:
        x: Input array of sample positions.
        n: Order of the spline (non-negative integer).

    Returns:
        Gaussian B-spline values at each element of *x*.
    """
    return _signal.gauss_spline(x, n)


def cubic(x):
    """Return a cubic B-spline basis function.

    Evaluates the third-order (cubic) B-spline at the points in *x*.

    Args:
        x: Input array of sample positions.

    Returns:
        Cubic B-spline values at each element of *x*.
    """
    return _signal.cubic(x)


def quadratic(x):
    """Return a quadratic B-spline basis function.

    Evaluates the second-order (quadratic) B-spline at the points in *x*.

    Args:
        x: Input array of sample positions.

    Returns:
        Quadratic B-spline values at each element of *x*.
    """
    return _signal.quadratic(x)

File: bindings/lua/insight/signal/bsplines.lua
--- B-spline basis functions.
-- Provides Gaussian, cubic, and quadratic B-spline functions.
-- @module insight.signal.bsplines

local native = require("_insight")
local sig = native.signal

local M = {}

--- Return a Gaussian B-spline basis function.
-- @int n Order of the spline.
-- @number x Evaluation points.
-- @treturn Array B-spline values.
function M.gauss_spline(n, x)
  return sig.gauss_spline(n, x)
end

--- Return a cubic B-spline.
-- @number x Evaluation points.
-- @treturn Array Cubic B-spline values.
function M.cubic(x)
  return sig.cubic(x)
end

--- Return a quadratic B-spline.
-- @number x Evaluation points.
-- @treturn Array Quadratic B-spline values.
function M.quadratic(x)
  return sig.quadratic(x)
end

return M
